### PR TITLE
idr0021 import for all workflow

### PR DESCRIPTION
**Pull request by pwalczysko**
This PR is pushing the enabling scripts to

   - import idr0021 for all (50) users on outreach (even when and if the project, dataset and images are already there for another or the same user for which the import loop is importing at that moment)
   - create a new project and link all datasets specified (by passing ``-d DATASET1 -d DATASET2 -d DATASET3 ...`` to it (even if there are more datasets with the same name for that user, the script should take the one created last and link it to the new project)
   - (meant as a follow-up option, the idr0021 workflow should be covered by the two steps above) link a dataset to  an existing project. This should work even in case there are 2 or more such projects or dtatasets for a given user, it should simply use the most recent project and dataset of that name. 

ToDo: Loop also for all users through the script which adds Map Annotations from IDR to the newly imported idr0021 projects

cc @jburel @sbesson @dominikl

### CODE REVIEW

**Comment by pwalczysko** [comment]
cc @mtbc - maybe a brilliant hql query could fix the Qustion from the header of this PR

**Comment by pwalczysko** [comment]
@mtbc says 

```
So, what you can do is before the query add `params.page(0, 1)` to get no more than one result back, then after the AND you end the query with `ORDER BY id DESC` to get only the latest-created.
```

**Comment by pwalczysko** [comment]
Thanks a lot to @mtbc and @manics for their help with the last three commits. 
This set of scripts now does what I want (it imports and links the idr0021 set or any other set of data appropriately and with very few steps, for all 50 users).

Editing the header of the PR

File: maintenance/scripts/link_dataset_to_existing_project.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+#
+#   You should have received a copy of the GNU General Public License along
+#   with this program; if not, write to the Free Software Foundation, Inc.,
+#   51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# ------------------------------------------------------------------------------
+
+"""
+This script looks for a specified project and links it to the specified Dataset
+for users user-1 through user-40.
```
Comment (jburel) [inline maintenance/scripts/link_dataset_to_existing_project.py]: range has changed

File: maintenance/scripts/link_dataset_to_new_project.py [hunk 2]
```diff
@@ -19,21 +19,30 @@
...
 from omero.model import ProjectDatasetLinkI
 from omero.model import ProjectI
 from omero.rtypes import rstring
 
 
-def run(password, project_name, dataset_name, host, port):
+def run(password, project_name, dataset_names, host, port):
 
-    for user_number in range(1, 41):
+    for user_number in range(1, 3):
```
Comment (jburel) [inline maintenance/scripts/link_dataset_to_new_project.py]: 3? this will need to be reverted

**Comment by pwalczysko** [comment]
@jburel the last commit should solve the new range of users for all edited scripts

**Comment by jburel** [comment]
The other scripts will need to be reviewed and updated if required but that should be done in a follow-up PR

**Comment by pwalczysko** [comment]
>The other scripts will need to be reviewed and updated if required but that should be done in a follow-up PR

Follow-up PR see #57

**Comment by joshmoore** [comment]
Longer-term also see my follow-up comment from https://github.com/ome/training-scripts/pull/57#issuecomment-457182645, but :+1:

(There are no automated tests for these scripts, correct?)

**Comment by pwalczysko** [comment]
>(There are no automated tests for these scripts, correct?)

Correct

**Comment by jburel** [comment]
merging so we can move forward with the prep.
This PR now conflicts with other work

**Pull request merged**